program/src/payload.rs

```diff
@@ -27,32 +24,96 @@ impl LeafInfo {
     }
 }
 
+#[repr(C)]
+#[cfg_attr(feature = "serde-feature", derive(Serialize, Deserialize))]
```

Comment:
Looks like the `serde` derives are still feature gated, but you removed the feature gating on the import in payload.rs, if this is what you want, that's fine, just making sure.